src/DracoCoffeeManager.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace Drupal\draco_coffee;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\State\StateInterface;
+
+/**
+ * DracoCoffeeManager service.
+ */
+class DracoCoffeeManager {
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * The Entity Type Manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The State API service.
+   *
+   * @var \Drupal\Core\State\StateInterface
+   */
+  protected $state;
+
+  /**
+   * The time service.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $time;
+
+  /**
+   * Constructs a DracoCoffeeManager object.
+   *
+   * @param \Drupal\draco_coffee\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface
+   *   The Entity Type Manager service.
+   * @param \Drupal\Core\State\StateInterface $state
+   *   The State API service.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The Time service.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, EntityTypeManagerInterface $entity_type_manager, StateInterface $state, TimeInterface $time) {
+    $this->configFactory = $config_factory;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->state = $state;
+    $this->time = $time;
+  }
+
+  /**
+   * Cron implementation.
+   */
+  public function cron() {
+    $config = $this->configFactory->get('draco_coffee.settings');
+    if (empty($this->state->get('draco_coffee.start')) ||
+      empty($config->get('role')) ||
+      empty($config->get('pots')) ||
+      empty($config->get('interval'))) {
+      return;
+    }
+
+    if ($this->isRefillNeeded()) {
+      $this->setBarista();
+      $this->increasePotCounter();
+
+    }
+    else {
+      $this->clear();
+    }
+    \Drupal::service('cache_tags.invalidator')->invalidateTags(['draco_coffee:state']);
```

Comment:
Inject this?

Reply:
Moved this to the DracoCoffeePot service.